// stylis/src/Parser.js
import { RULESET as F, DECLARATION as G, COMMENT as H } from "./Enum.js";
import { strlen as l, append as L, charat as I, replace as M, indexof as J, abs as q, sizeof as K, substr as O, trim as P, from as B } from "./Utility.js";
import { dealloc as Q, alloc as V, next as R, peek as x, delimit as y, identifier as W, commenter as X, escaping as Y, whitespace as Z, node as D, char as _, prev as $, caret as z } from "./Tokenizer.js";
function ca(e) {
  return Q(N("", null, null, null, [""], e = V(e), 0, [0], e));
}
function N(e, v, n, c, s, E, C, t, k) {
  for (var o = 0, h = 0, f = C, T = 0, d = 0, g = 0, i = 1, A = 1, b = 1, r = 0, p = "", u = s, w = E, m = c, a = p; A; )
    switch (g = r, r = R()) {
      // (
      case 40:
        if (g != 108 && I(a, f - 1) == 58) {
          J(a += M(y(r), "&", "&\f"), "&\f", q(o ? t[o - 1] : 0)) != -1 && (b = -1);
          break;
        }
      // " ' [
      case 34:
      case 39:
      case 91:
        a += y(r);
        break;
      // \t \n \r \s
      case 9:
      case 10:
      case 13:
      case 32:
        a += Z(g);
        break;
      // \
      case 92:
        a += Y(z() - 1, 7);
        continue;
      // /
      case 47:
        switch (x()) {
          case 42:
          case 47:
            L(j(X(R(), z()), v, n, k), k);
            break;
          default:
            a += "/";
        }
        break;
      // {
      case 123 * i:
        t[o++] = l(a) * b;
      // } ; \0
      case 125 * i:
      case 59:
      case 0:
        switch (r) {
          // \0 }
          case 0:
          case 125:
            A = 0;
          // ;
          case 59 + h:
            b == -1 && (a = M(a, /\f/g, "")), d > 0 && l(a) - f && L(d > 32 ? U(a + ";", c, n, f - 1, k) : U(M(a, " ", "") + ";", c, n, f - 2, k), k);
            break;
          // @ ;
          case 59:
            a += ";";
          // { rule/at-rule
          default:
            if (L(m = S(a, v, n, o, h, s, t, p, u = [], w = [], f, E), E), r === 123)
              if (h === 0)
                N(a, v, m, m, u, E, f, t, w);
              else
                switch (T === 99 && I(a, 3) === 110 ? 100 : T) {
                  // d l m s
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    N(e, m, m, c && L(S(e, m, m, 0, 0, s, t, p, s, u = [], f, w), w), s, w, f, t, c ? u : w);
                    break;
                  default:
                    N(a, m, m, m, [""], w, 0, t, w);
                }
        }
        o = h = d = 0, i = b = 1, p = a = "", f = C;
        break;
      // :
      case 58:
        f = 1 + l(a), d = g;
      default:
        if (i < 1) {
          if (r == 123)
            --i;
          else if (r == 125 && i++ == 0 && $() == 125)
            continue;
        }
        switch (a += B(r), r * i) {
          // &
          case 38:
            b = h > 0 ? 1 : (a += "\f", -1);
            break;
          // ,
          case 44:
            t[o++] = (l(a) - 1) * b, b = 1;
            break;
          // @
          case 64:
            x() === 45 && (a += y(R())), T = x(), h = f = l(p = a += W(z())), r++;
            break;
          // -
          case 45:
            g === 45 && l(a) == 2 && (i = 0);
        }
    }
  return E;
}
function S(e, v, n, c, s, E, C, t, k, o, h, f) {
  for (var T = s - 1, d = s === 0 ? E : [""], g = K(d), i = 0, A = 0, b = 0; i < c; ++i)
    for (var r = 0, p = O(e, T + 1, T = q(A = C[i])), u = e; r < g; ++r)
      (u = P(A > 0 ? d[r] + " " + p : M(p, /&\f/g, d[r]))) && (k[b++] = u);
  return D(e, v, n, s === 0 ? F : t, k, o, h, f);
}
function j(e, v, n, c) {
  return D(e, v, n, H, B(_()), O(e, 2, -2), 0, c);
}
function U(e, v, n, c, s) {
  return D(e, v, n, G, O(e, 0, c), O(e, c + 1, -1), c, s);
}
export {
  j as comment,
  ca as compile,
  U as declaration,
  N as parse,
  S as ruleset
};
