// styled-components/dist/styled-components.browser.esm.js
import { __assign as E, __spreadArray as x } from "../../tslib/tslib.es6.js";
import it from "../../@emotion/is-prop-valid/dist/emotion-is-prop-valid.esm.js";
import Y, { useDebugValue as Ee, createElement as at, useRef as ct, useContext as ut } from "react";
import lt from "../../@emotion/unitless/dist/emotion-unitless.esm.js";
import { RULESET as pt } from "../../stylis/src/Enum.js";
import { prefixer as dt, middleware as ht, rulesheet as ft } from "../../stylis/src/Middleware.js";
import { stringify as mt, serialize as yt } from "../../stylis/src/Serializer.js";
import { compile as vt } from "../../stylis/src/Parser.js";
var T = typeof process < "u" && process.env !== void 0 && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", ze = "active", $e = "data-styled-version", ne = "6.1.19", ye = `/*!sc*/
`, ee = typeof window < "u" && typeof document < "u", gt = !!(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), be = /invalid hook call/i, U = /* @__PURE__ */ new Set(), St = function(t, e) {
  if (process.env.NODE_ENV !== "production") {
    var n = e ? ' with the id of "'.concat(e, '"') : "", r = "The component ".concat(t).concat(n, ` has been created dynamically.
`) + `You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.
See https://styled-components.com/docs/basics#define-styled-components-outside-of-the-render-method for more info.
`, o = console.error;
    try {
      var s = !0;
      console.error = function(i) {
        for (var c = [], a = 1; a < arguments.length; a++) c[a - 1] = arguments[a];
        be.test(i) ? (s = !1, U.delete(r)) : o.apply(void 0, x([i], c, !1));
      }, ct(), s && !U.has(r) && (console.warn(r), U.add(r));
    } catch (i) {
      be.test(i.message) && U.delete(r);
    } finally {
      console.error = o;
    }
  }
}, re = Object.freeze([]), V = Object.freeze({});
function wt(t, e, n) {
  return n === void 0 && (n = V), t.theme !== n.theme && t.theme || e || n.theme;
}
var pe = /* @__PURE__ */ new Set(["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "tr", "track", "u", "ul", "use", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "tspan"]), Et = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, bt = /(^-|-$)/g;
function Ne(t) {
  return t.replace(Et, "-").replace(bt, "");
}
var Nt = /(a)(d)/gi, K = 52, _e = function(t) {
  return String.fromCharCode(t + (t > 25 ? 39 : 97));
};
function de(t) {
  var e, n = "";
  for (e = Math.abs(t); e > K; e = e / K | 0) n = _e(e % K) + n;
  return (_e(e % K) + n).replace(Nt, "$1-$2");
}
var ce, Fe = 5381, D = function(t, e) {
  for (var n = e.length; n; ) t = 33 * t ^ e.charCodeAt(--n);
  return t;
}, Be = function(t) {
  return D(Fe, t);
};
function _t(t) {
  return de(Be(t) >>> 0);
}
function Me(t) {
  return process.env.NODE_ENV !== "production" && typeof t == "string" && t || t.displayName || t.name || "Component";
}
function ue(t) {
  return typeof t == "string" && (process.env.NODE_ENV === "production" || t.charAt(0) === t.charAt(0).toLowerCase());
}
var Le = typeof Symbol == "function" && Symbol.for, Ge = Le ? Symbol.for("react.memo") : 60115, Ct = Le ? Symbol.for("react.forward_ref") : 60112, At = { childContextTypes: !0, contextType: !0, contextTypes: !0, defaultProps: !0, displayName: !0, getDefaultProps: !0, getDerivedStateFromError: !0, getDerivedStateFromProps: !0, mixins: !0, propTypes: !0, type: !0 }, Pt = { name: !0, length: !0, prototype: !0, caller: !0, callee: !0, arguments: !0, arity: !0 }, Ye = { $$typeof: !0, compare: !0, defaultProps: !0, displayName: !0, propTypes: !0, type: !0 }, It = ((ce = {})[Ct] = { $$typeof: !0, render: !0, defaultProps: !0, displayName: !0, propTypes: !0 }, ce[Ge] = Ye, ce);
function Ce(t) {
  return ("type" in (e = t) && e.type.$$typeof) === Ge ? Ye : "$$typeof" in t ? It[t.$$typeof] : At;
  var e;
}
var Dt = Object.defineProperty, Ot = Object.getOwnPropertyNames, Ae = Object.getOwnPropertySymbols, Rt = Object.getOwnPropertyDescriptor, Tt = Object.getPrototypeOf, Pe = Object.prototype;
function qe(t, e, n) {
  if (typeof e != "string") {
    if (Pe) {
      var r = Tt(e);
      r && r !== Pe && qe(t, r, n);
    }
    var o = Ot(e);
    Ae && (o = o.concat(Ae(e)));
    for (var s = Ce(t), i = Ce(e), c = 0; c < o.length; ++c) {
      var a = o[c];
      if (!(a in Pt || n && n[a] || i && a in i || s && a in s)) {
        var f = Rt(e, a);
        try {
          Dt(t, a, f);
        } catch {
        }
      }
    }
  }
  return t;
}
function z(t) {
  return typeof t == "function";
}
function ve(t) {
  return typeof t == "object" && "styledComponentId" in t;
}
function O(t, e) {
  return t && e ? "".concat(t, " ").concat(e) : t || e || "";
}
function Ie(t, e) {
  if (t.length === 0) return "";
  for (var n = t[0], r = 1; r < t.length; r++) n += t[r];
  return n;
}
function $(t) {
  return t !== null && typeof t == "object" && t.constructor.name === Object.name && !("props" in t && t.$$typeof);
}
function he(t, e, n) {
  if (n === void 0 && (n = !1), !n && !$(t) && !Array.isArray(t)) return e;
  if (Array.isArray(e)) for (var r = 0; r < e.length; r++) t[r] = he(t[r], e[r]);
  else if ($(e)) for (var r in e) t[r] = he(t[r], e[r]);
  return t;
}
function ge(t, e) {
  Object.defineProperty(t, "toString", { value: e });
}
var kt = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
`, 18: "ThemeProvider: Please make sure your useTheme hook is within a `<ThemeProvider>`" } : {};
function jt() {
  for (var t = [], e = 0; e < arguments.length; e++) t[e] = arguments[e];
  for (var n = t[0], r = [], o = 1, s = t.length; o < s; o += 1) r.push(t[o]);
  return r.forEach(function(i) {
    n = n.replace(/%[a-z]/, i);
  }), n;
}
function F(t) {
  for (var e = [], n = 1; n < arguments.length; n++) e[n - 1] = arguments[n];
  return process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/errors.md#".concat(t, " for more information.").concat(e.length > 0 ? " Args: ".concat(e.join(", ")) : "")) : new Error(jt.apply(void 0, x([kt[t]], e, !1)).trim());
}
var xt = (function() {
  function t(e) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = e;
  }
  return t.prototype.indexOfGroup = function(e) {
    for (var n = 0, r = 0; r < e; r++) n += this.groupSizes[r];
    return n;
  }, t.prototype.insertRules = function(e, n) {
    if (e >= this.groupSizes.length) {
      for (var r = this.groupSizes, o = r.length, s = o; e >= s; ) if ((s <<= 1) < 0) throw F(16, "".concat(e));
      this.groupSizes = new Uint32Array(s), this.groupSizes.set(r), this.length = s;
      for (var i = o; i < s; i++) this.groupSizes[i] = 0;
    }
    for (var c = this.indexOfGroup(e + 1), a = (i = 0, n.length); i < a; i++) this.tag.insertRule(c, n[i]) && (this.groupSizes[e]++, c++);
  }, t.prototype.clearGroup = function(e) {
    if (e < this.length) {
      var n = this.groupSizes[e], r = this.indexOfGroup(e), o = r + n;
      this.groupSizes[e] = 0;
      for (var s = r; s < o; s++) this.tag.deleteRule(r);
    }
  }, t.prototype.getGroup = function(e) {
    var n = "";
    if (e >= this.length || this.groupSizes[e] === 0) return n;
    for (var r = this.groupSizes[e], o = this.indexOfGroup(e), s = o + r, i = o; i < s; i++) n += "".concat(this.tag.getRule(i)).concat(ye);
    return n;
  }, t;
})(), Vt = 1 << 30, J = /* @__PURE__ */ new Map(), te = /* @__PURE__ */ new Map(), Q = 1, X = function(t) {
  if (J.has(t)) return J.get(t);
  for (; te.has(Q); ) Q++;
  var e = Q++;
  if (process.env.NODE_ENV !== "production" && ((0 | e) < 0 || e > Vt)) throw F(16, "".concat(e));
  return J.set(t, e), te.set(e, t), e;
}, zt = function(t, e) {
  Q = e + 1, J.set(t, e), te.set(e, t);
}, $t = "style[".concat(T, "][").concat($e, '="').concat(ne, '"]'), Ft = new RegExp("^".concat(T, '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)')), Bt = function(t, e, n) {
  for (var r, o = n.split(","), s = 0, i = o.length; s < i; s++) (r = o[s]) && t.registerName(e, r);
}, Mt = function(t, e) {
  for (var n, r = ((n = e.textContent) !== null && n !== void 0 ? n : "").split(ye), o = [], s = 0, i = r.length; s < i; s++) {
    var c = r[s].trim();
    if (c) {
      var a = c.match(Ft);
      if (a) {
        var f = 0 | parseInt(a[1], 10), d = a[2];
        f !== 0 && (zt(d, f), Bt(t, d, a[3]), t.getTag().insertRules(f, o)), o.length = 0;
      } else o.push(c);
    }
  }
}, De = function(t) {
  for (var e = document.querySelectorAll($t), n = 0, r = e.length; n < r; n++) {
    var o = e[n];
    o && o.getAttribute(T) !== ze && (Mt(t, o), o.parentNode && o.parentNode.removeChild(o));
  }
};
function Lt() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}
var He = function(t) {
  var e = document.head, n = t || e, r = document.createElement("style"), o = (function(c) {
    var a = Array.from(c.querySelectorAll("style[".concat(T, "]")));
    return a[a.length - 1];
  })(n), s = o !== void 0 ? o.nextSibling : null;
  r.setAttribute(T, ze), r.setAttribute($e, ne);
  var i = Lt();
  return i && r.setAttribute("nonce", i), n.insertBefore(r, s), r;
}, Gt = (function() {
  function t(e) {
    this.element = He(e), this.element.appendChild(document.createTextNode("")), this.sheet = (function(n) {
      if (n.sheet) return n.sheet;
      for (var r = document.styleSheets, o = 0, s = r.length; o < s; o++) {
        var i = r[o];
        if (i.ownerNode === n) return i;
      }
      throw F(17);
    })(this.element), this.length = 0;
  }
  return t.prototype.insertRule = function(e, n) {
    try {
      return this.sheet.insertRule(n, e), this.length++, !0;
    } catch {
      return !1;
    }
  }, t.prototype.deleteRule = function(e) {
    this.sheet.deleteRule(e), this.length--;
  }, t.prototype.getRule = function(e) {
    var n = this.sheet.cssRules[e];
    return n && n.cssText ? n.cssText : "";
  }, t;
})(), Yt = (function() {
  function t(e) {
    this.element = He(e), this.nodes = this.element.childNodes, this.length = 0;
  }
  return t.prototype.insertRule = function(e, n) {
    if (e <= this.length && e >= 0) {
      var r = document.createTextNode(n);
      return this.element.insertBefore(r, this.nodes[e] || null), this.length++, !0;
    }
    return !1;
  }, t.prototype.deleteRule = function(e) {
    this.element.removeChild(this.nodes[e]), this.length--;
  }, t.prototype.getRule = function(e) {
    return e < this.length ? this.nodes[e].textContent : "";
  }, t;
})(), qt = (function() {
  function t(e) {
    this.rules = [], this.length = 0;
  }
  return t.prototype.insertRule = function(e, n) {
    return e <= this.length && (this.rules.splice(e, 0, n), this.length++, !0);
  }, t.prototype.deleteRule = function(e) {
    this.rules.splice(e, 1), this.length--;
  }, t.prototype.getRule = function(e) {
    return e < this.length ? this.rules[e] : "";
  }, t;
})(), Oe = ee, Ht = { isServer: !ee, useCSSOMInjection: !gt }, We = (function() {
  function t(e, n, r) {
    e === void 0 && (e = V), n === void 0 && (n = {});
    var o = this;
    this.options = E(E({}, Ht), e), this.gs = n, this.names = new Map(r), this.server = !!e.isServer, !this.server && ee && Oe && (Oe = !1, De(this)), ge(this, function() {
      return (function(s) {
        for (var i = s.getTag(), c = i.length, a = "", f = function(h) {
          var u = (function(b) {
            return te.get(b);
          })(h);
          if (u === void 0) return "continue";
          var p = s.names.get(u), v = i.getGroup(h);
          if (p === void 0 || !p.size || v.length === 0) return "continue";
          var _ = "".concat(T, ".g").concat(h, '[id="').concat(u, '"]'), P = "";
          p !== void 0 && p.forEach(function(b) {
            b.length > 0 && (P += "".concat(b, ","));
          }), a += "".concat(v).concat(_, '{content:"').concat(P, '"}').concat(ye);
        }, d = 0; d < c; d++) f(d);
        return a;
      })(o);
    });
  }
  return t.registerId = function(e) {
    return X(e);
  }, t.prototype.rehydrate = function() {
    !this.server && ee && De(this);
  }, t.prototype.reconstructWithOptions = function(e, n) {
    return n === void 0 && (n = !0), new t(E(E({}, this.options), e), this.gs, n && this.names || void 0);
  }, t.prototype.allocateGSInstance = function(e) {
    return this.gs[e] = (this.gs[e] || 0) + 1;
  }, t.prototype.getTag = function() {
    return this.tag || (this.tag = (e = (function(n) {
      var r = n.useCSSOMInjection, o = n.target;
      return n.isServer ? new qt(o) : r ? new Gt(o) : new Yt(o);
    })(this.options), new xt(e)));
    var e;
  }, t.prototype.hasNameForId = function(e, n) {
    return this.names.has(e) && this.names.get(e).has(n);
  }, t.prototype.registerName = function(e, n) {
    if (X(e), this.names.has(e)) this.names.get(e).add(n);
    else {
      var r = /* @__PURE__ */ new Set();
      r.add(n), this.names.set(e, r);
    }
  }, t.prototype.insertRules = function(e, n, r) {
    this.registerName(e, n), this.getTag().insertRules(X(e), r);
  }, t.prototype.clearNames = function(e) {
    this.names.has(e) && this.names.get(e).clear();
  }, t.prototype.clearRules = function(e) {
    this.getTag().clearGroup(X(e)), this.clearNames(e);
  }, t.prototype.clearTag = function() {
    this.tag = void 0;
  }, t;
})(), Wt = /&/g, Ut = /^\s*\/\/.*$/gm;
function Ue(t, e) {
  return t.map(function(n) {
    return n.type === "rule" && (n.value = "".concat(e, " ").concat(n.value), n.value = n.value.replaceAll(",", ",".concat(e, " ")), n.props = n.props.map(function(r) {
      return "".concat(e, " ").concat(r);
    })), Array.isArray(n.children) && n.type !== "@keyframes" && (n.children = Ue(n.children, e)), n;
  });
}
function Kt(t) {
  var e, n, r, o = V, s = o.options, i = s === void 0 ? V : s, c = o.plugins, a = c === void 0 ? re : c, f = function(u, p, v) {
    return v.startsWith(n) && v.endsWith(n) && v.replaceAll(n, "").length > 0 ? ".".concat(e) : u;
  }, d = a.slice();
  d.push(function(u) {
    u.type === pt && u.value.includes("&") && (u.props[0] = u.props[0].replace(Wt, n).replace(r, f));
  }), i.prefix && d.push(dt), d.push(mt);
  var h = function(u, p, v, _) {
    p === void 0 && (p = ""), v === void 0 && (v = ""), _ === void 0 && (_ = "&"), e = _, n = p, r = new RegExp("\\".concat(n, "\\b"), "g");
    var P = u.replace(Ut, ""), b = vt(v || p ? "".concat(v, " ").concat(p, " { ").concat(P, " }") : P);
    i.namespace && (b = Ue(b, i.namespace));
    var B = [];
    return yt(b, ht(d.concat(ft(function(m) {
      return B.push(m);
    })))), B;
  };
  return h.hash = a.length ? a.reduce(function(u, p) {
    return p.name || F(15), D(u, p.name);
  }, Fe).toString() : "", h;
}
var Xt = new We(), fe = Kt(), Ke = Y.createContext({ shouldForwardProp: void 0, styleSheet: Xt, stylis: fe });
Ke.Consumer;
Y.createContext(void 0);
function Re() {
  return ut(Ke);
}
var Te = (function() {
  function t(e, n) {
    var r = this;
    this.inject = function(o, s) {
      s === void 0 && (s = fe);
      var i = r.name + s.hash;
      o.hasNameForId(r.id, i) || o.insertRules(r.id, i, s(r.rules, i, "@keyframes"));
    }, this.name = e, this.id = "sc-keyframes-".concat(e), this.rules = n, ge(this, function() {
      throw F(12, String(r.name));
    });
  }
  return t.prototype.getName = function(e) {
    return e === void 0 && (e = fe), this.name + e.hash;
  }, t;
})(), Zt = function(t) {
  return t >= "A" && t <= "Z";
};
function ke(t) {
  for (var e = "", n = 0; n < t.length; n++) {
    var r = t[n];
    if (n === 1 && r === "-" && t[0] === "-") return t;
    Zt(r) ? e += "-" + r.toLowerCase() : e += r;
  }
  return e.startsWith("ms-") ? "-" + e : e;
}
var Xe = function(t) {
  return t == null || t === !1 || t === "";
}, Ze = function(t) {
  var e, n, r = [];
  for (var o in t) {
    var s = t[o];
    t.hasOwnProperty(o) && !Xe(s) && (Array.isArray(s) && s.isCss || z(s) ? r.push("".concat(ke(o), ":"), s, ";") : $(s) ? r.push.apply(r, x(x(["".concat(o, " {")], Ze(s), !1), ["}"], !1)) : r.push("".concat(ke(o), ": ").concat((e = o, (n = s) == null || typeof n == "boolean" || n === "" ? "" : typeof n != "number" || n === 0 || e in lt || e.startsWith("--") ? String(n).trim() : "".concat(n, "px")), ";")));
  }
  return r;
};
function R(t, e, n, r) {
  if (Xe(t)) return [];
  if (ve(t)) return [".".concat(t.styledComponentId)];
  if (z(t)) {
    if (!z(s = t) || s.prototype && s.prototype.isReactComponent || !e) return [t];
    var o = t(e);
    return process.env.NODE_ENV === "production" || typeof o != "object" || Array.isArray(o) || o instanceof Te || $(o) || o === null || console.error("".concat(Me(t), " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.")), R(o, e, n, r);
  }
  var s;
  return t instanceof Te ? n ? (t.inject(n, r), [t.getName(r)]) : [t] : $(t) ? Ze(t) : Array.isArray(t) ? Array.prototype.concat.apply(re, t.map(function(i) {
    return R(i, e, n, r);
  })) : [t.toString()];
}
function Jt(t) {
  for (var e = 0; e < t.length; e += 1) {
    var n = t[e];
    if (z(n) && !ve(n)) return !1;
  }
  return !0;
}
var Qt = Be(ne), en = (function() {
  function t(e, n, r) {
    this.rules = e, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (r === void 0 || r.isStatic) && Jt(e), this.componentId = n, this.baseHash = D(Qt, n), this.baseStyle = r, We.registerId(n);
  }
  return t.prototype.generateAndInjectStyles = function(e, n, r) {
    var o = this.baseStyle ? this.baseStyle.generateAndInjectStyles(e, n, r) : "";
    if (this.isStatic && !r.hash) if (this.staticRulesId && n.hasNameForId(this.componentId, this.staticRulesId)) o = O(o, this.staticRulesId);
    else {
      var s = Ie(R(this.rules, e, n, r)), i = de(D(this.baseHash, s) >>> 0);
      if (!n.hasNameForId(this.componentId, i)) {
        var c = r(s, ".".concat(i), void 0, this.componentId);
        n.insertRules(this.componentId, i, c);
      }
      o = O(o, i), this.staticRulesId = i;
    }
    else {
      for (var a = D(this.baseHash, r.hash), f = "", d = 0; d < this.rules.length; d++) {
        var h = this.rules[d];
        if (typeof h == "string") f += h, process.env.NODE_ENV !== "production" && (a = D(a, h));
        else if (h) {
          var u = Ie(R(h, e, n, r));
          a = D(a, u + d), f += u;
        }
      }
      if (f) {
        var p = de(a >>> 0);
        n.hasNameForId(this.componentId, p) || n.insertRules(this.componentId, p, r(f, ".".concat(p), void 0, this.componentId)), o = O(o, p);
      }
    }
    return o;
  }, t;
})(), Je = Y.createContext(void 0);
Je.Consumer;
var le = {}, je = /* @__PURE__ */ new Set();
function tn(t, e, n) {
  var r = ve(t), o = t, s = !ue(t), i = e.attrs, c = i === void 0 ? re : i, a = e.componentId, f = a === void 0 ? (function(y, g) {
    var l = typeof y != "string" ? "sc" : Ne(y);
    le[l] = (le[l] || 0) + 1;
    var S = "".concat(l, "-").concat(_t(ne + l + le[l]));
    return g ? "".concat(g, "-").concat(S) : S;
  })(e.displayName, e.parentComponentId) : a, d = e.displayName, h = d === void 0 ? (function(y) {
    return ue(y) ? "styled.".concat(y) : "Styled(".concat(Me(y), ")");
  })(t) : d, u = e.displayName && e.componentId ? "".concat(Ne(e.displayName), "-").concat(e.componentId) : e.componentId || f, p = r && o.attrs ? o.attrs.concat(c).filter(Boolean) : c, v = e.shouldForwardProp;
  if (r && o.shouldForwardProp) {
    var _ = o.shouldForwardProp;
    if (e.shouldForwardProp) {
      var P = e.shouldForwardProp;
      v = function(y, g) {
        return _(y, g) && P(y, g);
      };
    } else v = _;
  }
  var b = new en(n, u, r ? o.componentStyle : void 0);
  function B(y, g) {
    return (function(l, S, C) {
      var k = l.attrs, et = l.componentStyle, tt = l.defaultProps, nt = l.foldedComponentIds, Se = l.styledComponentId, rt = l.target, ot = Y.useContext(Je), st = Re(), oe = l.shouldForwardProp || st.shouldForwardProp;
      process.env.NODE_ENV !== "production" && Ee(Se);
      var we = wt(S, ot, tt) || V, N = (function(q, G, H) {
        for (var j, I = E(E({}, G), { className: void 0, theme: H }), ae = 0; ae < q.length; ae += 1) {
          var W = z(j = q[ae]) ? j(I) : j;
          for (var A in W) I[A] = A === "className" ? O(I[A], W[A]) : A === "style" ? E(E({}, I[A]), W[A]) : W[A];
        }
        return G.className && (I.className = O(I.className, G.className)), I;
      })(k, S, we), M = N.as || rt, L = {};
      for (var w in N) N[w] === void 0 || w[0] === "$" || w === "as" || w === "theme" && N.theme === we || (w === "forwardedAs" ? L.as = N.forwardedAs : oe && !oe(w, M) || (L[w] = N[w], oe || process.env.NODE_ENV !== "development" || it(w) || je.has(w) || !pe.has(M) || (je.add(w), console.warn('styled-components: it looks like an unknown prop "'.concat(w, '" is being sent through to the DOM, which will likely trigger a React console error. If you would like automatic filtering of unknown props, you can opt-into that behavior via `<StyleSheetManager shouldForwardProp={...}>` (connect an API like `@emotion/is-prop-valid`) or consider using transient props (`$` prefix for automatic filtering.)')))));
      var se = (function(q, G) {
        var H = Re(), j = q.generateAndInjectStyles(G, H.styleSheet, H.stylis);
        return process.env.NODE_ENV !== "production" && Ee(j), j;
      })(et, N);
      process.env.NODE_ENV !== "production" && l.warnTooManyClasses && l.warnTooManyClasses(se);
      var ie = O(nt, Se);
      return se && (ie += " " + se), N.className && (ie += " " + N.className), L[ue(M) && !pe.has(M) ? "class" : "className"] = ie, C && (L.ref = C), at(M, L);
    })(m, y, g);
  }
  B.displayName = h;
  var m = Y.forwardRef(B);
  return m.attrs = p, m.componentStyle = b, m.displayName = h, m.shouldForwardProp = v, m.foldedComponentIds = r ? O(o.foldedComponentIds, o.styledComponentId) : "", m.styledComponentId = u, m.target = r ? o.target : t, Object.defineProperty(m, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(y) {
    this._foldedDefaultProps = r ? (function(g) {
      for (var l = [], S = 1; S < arguments.length; S++) l[S - 1] = arguments[S];
      for (var C = 0, k = l; C < k.length; C++) he(g, k[C], !0);
      return g;
    })({}, o.defaultProps, y) : y;
  } }), process.env.NODE_ENV !== "production" && (St(h, u), m.warnTooManyClasses = /* @__PURE__ */ (function(y, g) {
    var l = {}, S = !1;
    return function(C) {
      if (!S && (l[C] = !0, Object.keys(l).length >= 200)) {
        var k = g ? ' with the id of "'.concat(g, '"') : "";
        console.warn("Over ".concat(200, " classes were generated for component ").concat(y).concat(k, `.
`) + `Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), S = !0, l = {};
      }
    };
  })(h, u)), ge(m, function() {
    return ".".concat(m.styledComponentId);
  }), s && qe(m, t, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0 }), m;
}
function xe(t, e) {
  for (var n = [t[0]], r = 0, o = e.length; r < o; r += 1) n.push(e[r], t[r + 1]);
  return n;
}
var Ve = function(t) {
  return Object.assign(t, { isCss: !0 });
};
function nn(t) {
  for (var e = [], n = 1; n < arguments.length; n++) e[n - 1] = arguments[n];
  if (z(t) || $(t)) return Ve(R(xe(re, x([t], e, !0))));
  var r = t;
  return e.length === 0 && r.length === 1 && typeof r[0] == "string" ? R(r) : Ve(R(xe(r, e)));
}
function me(t, e, n) {
  if (n === void 0 && (n = V), !e) throw F(1, e);
  var r = function(o) {
    for (var s = [], i = 1; i < arguments.length; i++) s[i - 1] = arguments[i];
    return t(e, n, nn.apply(void 0, x([o], s, !1)));
  };
  return r.attrs = function(o) {
    return me(t, e, E(E({}, n), { attrs: Array.prototype.concat(n.attrs, o).filter(Boolean) }));
  }, r.withConfig = function(o) {
    return me(t, e, E(E({}, n), o));
  }, r;
}
var Qe = function(t) {
  return me(tn, t);
}, rn = Qe;
pe.forEach(function(t) {
  rn[t] = Qe(t);
});
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`);
var Z = "__sc-".concat(T, "__");
process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window[Z] || (window[Z] = 0), window[Z] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window[Z] += 1);
export {
  Ke as StyleSheetContext,
  Je as ThemeContext,
  nn as css,
  rn as default,
  ve as isStyledComponent,
  rn as styled,
  ne as version
};
